security: broken test `//pkg/security:security_test` in Bazel

Regressed by `e34411c29e1b9db0f50e14bbe302156a8ae77a6b`

```
W210805 16:49:39.662290 23 security/certificate_loader.go:339  [-] 1  bad filename /tmp/certs_test614838746/ca.foo.crt: CA certificate filename should match ca.crt
W210805 16:49:39.662409 23 security/certificate_loader.go:339  [-] 2  bad filename /tmp/certs_test614838746/client..crt: client certificate filename should match client.<user>.crt
W210805 16:49:39.662450 23 security/certificate_loader.go:339  [-] 3  bad filename /tmp/certs_test614838746/client.crt: client certificate filename should match client.<user>.crt
W210805 16:49:39.662485 23 security/certificate_loader.go:339  [-] 4  bad filename /tmp/certs_test614838746/cr..crt: unknown prefix "cr"
W210805 16:49:39.662521 23 security/certificate_loader.go:339  [-] 5  bad filename /tmp/certs_test614838746/node..crt: node certificate filename should match node.crt
W210805 16:49:39.662554 23 security/certificate_loader.go:339  [-] 6  bad filename /tmp/certs_test614838746/node.foo.crt: node certificate filename should match node.crt
W210805 16:49:39.664125 23 security/certificate_loader.go:357  [-] 7  error finding key for /tmp/certs_test614838746/client.root.crt: could not stat key file /tmp/certs_test614838746/client.root.key: stat /tmp/certs_test614838746/client.root.key: no such file or directory
W210805 16:49:39.664271 23 security/certificate_loader.go:357  [-] 8  error finding key for /tmp/certs_test614838746/node.crt: could not stat key file /tmp/certs_test614838746/node.key: stat /tmp/certs_test614838746/node.key: no such file or directory
W210805 16:49:39.664720 23 security/certificate_loader.go:357  [-] 9  error finding key for /tmp/certs_test614838746/node.crt: key file /tmp/certs_test614838746/node.key has permissions -rwx---r--, exceeds -rwxr-----
W210805 16:49:39.665044 23 security/certificate_loader.go:354  [-] 10  could not parse certificate for /tmp/certs_test614838746/client.root.crt: failed to validate certificate 0 in file client.root.crt: client certificate has principals ["notroot"], expected "root"
--- FAIL: TestNamingScheme (0.03s)
    certificate_loader_test.go:389: #3: mismatched error, expected: .* exceeds -rwx------
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/security_test.TestNamingScheme
          | 	github.com/cockroachdb/cockroach/pkg/security_test/certificate_loader_test.go:271
          | testing.tRunner
          | 	GOROOT/src/testing/testing.go:1193
          | runtime.goexit
          | 	GOROOT/src/runtime/asm_amd64.s:1371
        Wraps: (2) .* exceeds -rwx------
        Error types: (1) *withstack.withStack (2) *errutil.leafError, got <nil>
    certificate_loader_test.go:389: #3: mismatched error, expected: .* exceeds -rwx------
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/security_test.TestNamingScheme
          | 	github.com/cockroachdb/cockroach/pkg/security_test/certificate_loader_test.go:273
          | testing.tRunner
          | 	GOROOT/src/testing/testing.go:1193
          | runtime.goexit
          | 	GOROOT/src/runtime/asm_amd64.s:1371
        Wraps: (2) .* exceeds -rwx------
        Error types: (1) *withstack.withStack (2) *errutil.leafError, got key file /tmp/certs_test614838746/node.key has permissions -rwx---r--, exceeds -rwxr-----
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/security.checkFilePermissions
          | 	github.com/cockroachdb/cockroach/pkg/security/permission_check.go:27
          | github.com/cockroachdb/cockroach/pkg/security.(*CertificateLoader).findKey
          | 	github.com/cockroachdb/cockroach/pkg/security/certificate_loader.go:394
          | github.com/cockroachdb/cockroach/pkg/security.(*CertificateLoader).Load
          | 	github.com/cockroachdb/cockroach/pkg/security/certificate_loader.go:356
          | github.com/cockroachdb/cockroach/pkg/security_test.TestNamingScheme
          | 	github.com/cockroachdb/cockroach/pkg/security_test/certificate_loader_test.go:369
          | testing.tRunner
          | 	GOROOT/src/testing/testing.go:1193
          | runtime.goexit
          | 	GOROOT/src/runtime/asm_amd64.s:1371
        Wraps: (2) key file /tmp/certs_test614838746/node.key has permissions -rwx---r--, exceeds -rwxr-----
        Error types: (1) *withstack.withStack (2) *errutil.leafError
```

--- a/pkg/security/certificate_loader_test.go
+++ b/pkg/security/certificate_loader_test.go
@@ -167,6 +167,22 @@ func makeTestCert(
 func TestNamingScheme(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 
+	permissionRequirement := ".* exceeds -rwx------"
+
+	// by default we only allow certificate keys owned by the same user
+	// as the running process (0700). In some environments, such as
+	// Kubernetes, we do not have control over owning user for volume
+	// mounts (such as Secrets) which always mount as root, and in
+	// this case we allow the group read permission (0740), provided
+	// the process shares the same group ID and the owning user of the
+	// key is root.
+	//
+	// Because this test creates certificate keys, when run as root we
+	// we have the less stringent permission requirement.
+	if os.Getuid() == 0 {
+		permissionRequirement = "exceeds -rwxr-----"
+	}
+
 	fullKeyUsage := x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature
 	// Build a few certificates. These are barebones since we only need to check our custom validation,
 	// not chain verification.
@@ -263,14 +279,14 @@ func TestNamingScheme(t *testing.T) {
 				{"node.crt", 0777, goodNodeCert},
 				{"node.key", 0704, []byte{}},
 				{"client.root.crt", 0777, goodRootCert},
-				{"client.root.key", 0740, []byte{}},
+				{"client.root.key", 0704, []byte{}},
 			},
 			certs: []security.CertInfo{
 				{FileUsage: security.CAPem, Filename: "ca.crt", FileContents: caCert},
 				{FileUsage: security.ClientPem, Filename: "client.root.crt", Name: "root",
-					Error: errors.New(".* exceeds -rwx------")},
+					Error: errors.New(permissionRequirement)},
 				{FileUsage: security.NodePem, Filename: "node.crt",
-					Error: errors.New(".* exceeds -rwx------")},
+					Error: errors.New(permissionRequirement)},
 			},
 			skipWindows: true,
 		},

--- a/pkg/util/sysutil/acl_unix_test.go
+++ b/pkg/util/sysutil/acl_unix_test.go
@@ -23,7 +23,7 @@ import (
 )
 
 func TestGetFileACLInfo(t *testing.T) {
-	certsDir, err := ioutil.TempDir("", "certs_test")
+	certsDir, err := ioutil.TempDir("", "acl_test")
 	if err != nil {
 		t.Fatal(err)
 	}

